### PR TITLE
feat: show environments in a separate dropdown

### PR DESCRIPTION
**Pull request by YounixM**
#### Screenshots

NA

### CODE REVIEW

File: frontend/src/hooks/useResourceAttribute/ResourceProvider.tsx [hunk 1]
```diff
@@ -127,6 +131,31 @@ function ResourceProvider({ children }: Props): JSX.Element {
 		[optionsData.mode, send],
 	);
 
+	const handleEnvironmentChange = useCallback(
+		(environments: string[]): void => {
+			const staging = ['resource_deployment_environment', 'IN'];
+
+			console.log('value', environments);
```
Comment (Vikrant2520) [inline frontend/src/hooks/useResourceAttribute/ResourceProvider.tsx:138]: console

File: frontend/src/container/ResourceAttributesFilter/ResourceAttributesFilter.tsx [hunk 2]
```diff
@@ -22,60 +29,129 @@ function ResourceAttributesFilter({
...
+	useEffect(() => {
+		getEnvironmentTagKeys()
+			.then((tagKeys) => {
+				if (tagKeys && Array.isArray(tagKeys) && tagKeys.length > 0) {
+					getEnvironmentTagValues().then((tagValues) => {
+						setEnvironments(tagValues);
+					});
+				}
+			})
+			.finally(() => {});
```
Comment (Vikrant2520) [inline frontend/src/container/ResourceAttributesFilter/ResourceAttributesFilter.tsx:78]: is this required ?

**Review by Vikrant2520 — DISMISSED** [review]

**Review by Rajat-Dabade — APPROVED** [review]

**Comment by srikanthccv** [comment]
Functionally looks good to me. It would have been good have this propagated to explorer pages also but that can be taken up later.

File: frontend/src/container/ResourceAttributesFilter/ResourceAttributesFilter.tsx [hunk 3]
```diff
@@ -22,60 +29,129 @@ function ResourceAttributesFilter({
...
+	const [environments, setEnvironments] = useState<
+		SelectOption<string, string>[]
+	>([]);
+
+	const [selectedEnvironments, setSelectedEnvironments] = useState<string[]>([]);
+
+	const queriesExcludingEnvironment = useMemo(
+		() =>
+			queries.filter(
+				(query) => query.tagKey !== 'resource_deployment_environment',
```
Comment (Rajat-Dabade) [inline frontend/src/container/ResourceAttributesFilter/ResourceAttributesFilter.tsx:47]: NIT: `resource_deployment_environment` to a contant.

File: frontend/src/container/ResourceAttributesFilter/ResourceAttributesFilter.tsx [hunk 4]
```diff
@@ -22,60 +29,129 @@ function ResourceAttributesFilter({
...
 	const isEmpty = useMemo(
-		() => isResourceEmpty(queries, staging, selectedQuery),
-		[queries, selectedQuery, staging],
+		() => isResourceEmpty(queriesExcludingEnvironment, staging, selectedQuery),
+		[queriesExcludingEnvironment, selectedQuery, staging],
 	);
 
+	useEffect(() => {
+		const resourceDeploymentEnvironmentQuery = queries.filter(
+			(query) => query.tagKey === 'resource_deployment_environment',
```
Comment (Rajat-Dabade) [inline frontend/src/container/ResourceAttributesFilter/ResourceAttributesFilter.tsx:59]: Same here, to the constants.

**Review by Rajat-Dabade — COMMENTED** [review]
Light mode border is missing

Screenshot 2024-03-26 at 11 37 22 PM

**Review by srikanthccv — APPROVED** [review]
LGTM, tested the functionality.

File: frontend/src/hooks/useResourceAttribute/utils.ts [hunk 5]
```diff
@@ -132,6 +163,23 @@ export const GetTagValues = async (tagKey: string): Promise<IOption[]> => {
 
 export const createQuery = (
 	selectedItems: Array<string | string[]> = [],
+): IResourceAttribute | null => {
+	console.log('selectedItems', selectedItems);
```
Comment (Vikrant2520) [inline frontend/src/hooks/useResourceAttribute/utils.ts:167]: console.log

**Review by Vikrant2520 — APPROVED** [review]

File: frontend/src/container/ResourceAttributesFilter/ResourceAttributesFilter.tsx (hunk 3 above)
Comment (YounixM) [reply frontend/src/container/ResourceAttributesFilter/ResourceAttributesFilter.tsx:47]: Will pick this in another PR

File: frontend/src/container/ResourceAttributesFilter/ResourceAttributesFilter.tsx (hunk 4 above)
Comment (YounixM) [reply frontend/src/container/ResourceAttributesFilter/ResourceAttributesFilter.tsx:59]: Will pick this in another PR

File: frontend/src/hooks/useResourceAttribute/utils.ts (hunk 5 above)
Comment (YounixM) [reply frontend/src/hooks/useResourceAttribute/utils.ts:167]: Will pick this in another PR

**Pull request merged**